### PR TITLE
Small updates to the EmCom as SSL game

### PR DESCRIPTION
**Pull request by Slowika**
## Description
* README (by Roberto)
* Option to train on CIFAR (cmd argument, appropriate normalization)
* Sweep jsons with the ImageNet path
* use_larc: [true] 
* remove nest.txt

## Motivation and Context
Experiment 0 for the population game

### CODE REVIEW

File: egg/zoo/emcom_as_ssl/archs.py [hunk 1]
```diff
@@ -43,10 +47,10 @@ def get_vision_modules(
     if pretrain_vision:
         assert shared, "A pretrained not shared vision_module is a waste of memory. Please run with --shared set"
 
-    encoder, features_dim = get_resnet(encoder_arch, pretrain_vision)
+    encoder, features_dim = get_pretrained(encoder_arch, pretrain_vision)
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/archs.py:50]: Not all are pretrained. Should we call it get_vision_module?

File: egg/zoo/emcom_as_ssl/data.py [hunk 2]
```diff
@@ -86,7 +88,7 @@ def __init__(self, size: int, augmentation: bool = False, return_original_image:
         transformations.extend([
             transforms.ToTensor(),
             transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/data.py:90]: Should this be removed if we train with cifar?

File: egg/zoo/emcom_as_ssl/games.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 def build_vision_encoder(
     model_name: str = "resnet50",
     shared_vision: bool = False,
-    pretrain_vision: bool = False
+    pretrain_vision: bool = True
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/games.py:24]: I think this should be False by default

File: egg/zoo/emcom_as_ssl/sweeps/game_no_augmentations_shared_vision_trained.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    "lr": [2.4],
+    "random_seed": [111],
+    "shared_vision": [true],
+    "gs_temperature": [5.0],
+    "loss_type": ["xent"],
+    "validation_freq": [0],
+    "image_size": [32],
+    "model_name": ["resnet50"],
+    "dataset_name": ["cifar10"]
+}
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/sweeps/game_no_augmentations_shared_vision_trained.json:13]: I'd add cifar to the file name just to differentiate

File: egg/zoo/emcom_as_ssl/sweeps/pretrained_imagenet.json [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/sweeps/pretrained_imagenet.json:1]: I'd add non_shared to the name or something else a bit more informative

File: egg/zoo/emcom_as_ssl/sweeps/trained_imagenet.json [hunk 6]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/sweeps/trained_imagenet.json:1]: Same as previous file

File: requirements.txt [hunk 7]
```diff
@@ -8,3 +8,4 @@ dataclasses
 rich
 scikit-learn
 wandb
+submitit
```
Comment (robertodessi) [inline requirements.txt:11]: submitit is already here: https://github.com/facebookresearch/EGG/blob/master/nest.txt

We should just add to the readme that in order to use nest one should pip install nest.txt as well

File: tests/test_games_do_not_fail.py [hunk 8]
```diff
@@ -104,6 +104,11 @@ def test_language_bottleneck():
         "egg.zoo.language_bottleneck.guess_number.train", dict(n_epochs=1, vocab_size=5)
     )
 
+def test_emcom_as_ssl_cifar():
+    run_game(
+        "egg.zoo.emcom_as_ssl.train", dict(dataset_name='cifar10', n_epoch=1)
+    )
+
```
Comment (robertodessi) [inline tests/test_games_do_not_fail.py:111]: Have you tried this locally? It seems CI are failing with a timeout error (> 10mins). Maybe defaults are creating a big (image_size too large and hidden dims unnecessary big?) game?

File: egg/zoo/emcom_as_ssl/archs.py [hunk 9]
```diff
@@ -13,17 +13,21 @@
...
+        "densenet": torchvision.models.densenet161(pretrained=pretrained),
+        "inception": torchvision.models.inception_v3(pretrained=pretrained)
     }
-    if name not in resnets:
+    if name not in modules:
         raise KeyError(f"{name} is not a valid ResNet architecture")
 
-    model = resnets[name]
+    model = modules[name]
     n_features = model.fc.in_features
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/archs.py:31]: This wouldn't work with vgg. 

https://pytorch.org/tutorials/beginner/finetuning_torchvision_models_tutorial.html

File: egg/zoo/emcom_as_ssl/data.py [hunk 10]
```diff
@@ -78,12 +80,19 @@ def __init__(
             ]
         else:
             transformations = [transforms.Resize(size=(size, size))]
+            
+        if dataset_name = 'imagenet':
+            mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]
+        else:
+            mean=[0.5, 0.5, 0.5], std=[0.5, 0.5, 0.5]
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/data.py:87]: Should this be only for cifar? If in the future we want to add other datasets maybe we should leave this optional

File: egg/zoo/emcom_as_ssl/archs.py [hunk 11]
```diff
@@ -13,17 +13,21 @@
...
         "resnet50": torchvision.models.resnet50(pretrained=pretrained),
         "resnet101": torchvision.models.resnet101(pretrained=pretrained),
         "resnet152": torchvision.models.resnet152(pretrained=pretrained),
+        "vgg": torchvision.models.vgg16(pretrained=pretrained),
+        "densenet": torchvision.models.densenet161(pretrained=pretrained),
+        "inception": torchvision.models.inception_v3(pretrained=pretrained),
     }
-    if name not in resnets:
+    if name not in modules:
         raise KeyError(f"{name} is not a valid ResNet architecture")
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/archs.py:28]: removing resnet from the error message

File: egg/zoo/emcom_as_ssl/train.py [hunk 12]
```diff
@@ -30,12 +30,11 @@ def main(params):
...
     train_loader = get_dataloader(
         dataset_dir=opts.dataset_dir,
+        dataset_name=opts.dataset_name,
         image_size=opts.image_size,
         batch_size=opts.batch_size,
         num_workers=opts.num_workers,
-        use_augmentations=opts.use_augmentations,
         is_distributed=opts.distributed_context.is_distributed,
-        return_original_image=opts.return_original_image,
         seed=opts.random_seed,
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/train.py:38]: ?

File: egg/zoo/emcom_as_ssl/README.txt [hunk 13]
```diff
@@ -0,0 +1,58 @@
+INSTALLATION
+
+We added all the code for the experiments under EGG/egg/zoo/emcom_as_ssl.
+As a first step, EGG needs to be installed. Following the instructions found at https://github.com/facebookresearch/EGG, from the EGG root directory the following command must be executed:
+
+$ pip install --editable .
+
```
Comment (robertodessi) [inline egg/zoo/emcom_as_ssl/README.txt:7]: I'd remove this

File: egg/zoo/simclr/README.md [hunk 14]
```diff
@@ -9,5 +9,5 @@ python -m torch.distributed.launch --use_env --nproc_per_node=2 egg/zoo/simclr_r
 In `sweeps` there's a configuration that tries to reproduce the setup of the paper with a batch size of 2048 using 16 GPUs
 In can be launched calling nest nest from the root directory of EGG with:
 ```bash
-python egg/nest/nest.py --game egg.zoo.simclr.train --sweep egg/zoo/simclr_replica/sweeps/simclr.json --checkpoint_dir="replicate_simclr" --nodes=2 --tasks=8
+python egg/nest/nest.py --game egg.zoo.simclr.train --sweep egg/zoo/simclr_replica/extension_sweeps/simclr.json --checkpoint_dir="replicate_simclr" --nodes=2 --tasks=8
```
Comment (robertodessi) [inline egg/zoo/simclr/README.md:12]: ?

File: egg/zoo/simclr/README.md [hunk 15]
```diff
@@ -9,5 +9,5 @@ python -m torch.distributed.launch --use_env --nproc_per_node=2 egg/zoo/simclr_r
 In `sweeps` there's a configuration that tries to reproduce the setup of the paper with a batch size of 2048 using 16 GPUs
 In can be launched calling nest nest from the root directory of EGG with:
 ```bash
-python egg/nest/nest.py --game egg.zoo.simclr.train --sweep egg/zoo/simclr_replica/sweeps/simclr.json --checkpoint_dir="replicate_simclr" --nodes=2 --tasks=8
+python egg/nest/nest.py --game egg.zoo.simclr.train --sweep egg/zoo/simclr_replica/paper_sweeps/simclr.json --checkpoint_dir="replicate_simclr" --nodes=2 --tasks=8
```
Comment (robertodessi) [inline egg/zoo/simclr/README.md]: This is for a different game, isn't it?

Accidentally, it should be simclr rather than simclr_replica in the sweep path :D

File: egg/zoo/simclr/README.md [hunk 16]
```diff
@@ -9,5 +9,5 @@ python -m torch.distributed.launch --use_env --nproc_per_node=2 egg/zoo/simclr_r
 In `sweeps` there's a configuration that tries to reproduce the setup of the paper with a batch size of 2048 using 16 GPUs
 In can be launched calling nest nest from the root directory of EGG with:
 ```bash
-python egg/nest/nest.py --game egg.zoo.simclr.train --sweep egg/zoo/simclr_replica/sweeps/simclr.json --checkpoint_dir="replicate_simclr" --nodes=2 --tasks=8
+python egg/nest/nest.py --game egg.zoo.simclr.train --sweep egg/zoo/simclr_replica/paper_sweeps/simclr_augmentations_shared_vision.json --checkpoint_dir="replicate_simclr" --nodes=2 --tasks=8
```
Comment (robertodessi) [inline egg/zoo/simclr/README.md:12]: This is for a different game, isn't it?

Accidentally, it should be simclr only instead of simclr_replica :D
